src/LanguageServer/Protocol/Workspaces/ILspMiscellaneousFilesWorkspaceProvider.cs

```diff
@@ -15,6 +16,10 @@ internal interface ILspMiscellaneousFilesWorkspaceProvider : ILspService
     /// Returns the actual workspace that the documents are added to or removed from.
     /// </summary>
     Workspace Workspace { get; }
-    TextDocument? AddMiscellaneousDocument(DocumentUri uri, SourceText documentText, string languageId, ILspLogger logger);
-    void TryRemoveMiscellaneousDocument(DocumentUri uri, bool removeFromMetadataWorkspace);
+    /// <summary>
+    /// Adds a document to the workspace. Note that the implementation of this method should not depend on anything expensive such as RPC calls.
+    /// async is used here to allow taking locks asynchronously and "relatively fast" stuff like that.
+    /// </summary>
+    Task<TextDocument?> AddMiscellaneousDocumentAsync(DocumentUri uri, SourceText documentText, string languageId, ILspLogger logger);
+    ValueTask TryRemoveMiscellaneousDocumentAsync(DocumentUri uri, bool removeFromMetadataWorkspace);
```

Comment:
why is this 'try' but doesn't return a boolean?

Reply:
I did not introduce the 'Try' prefix here. But I believe the goal of it is to indicate that it's fine to call this for a document which was already removed from the workspace.

The LSP queue does this frequently. Whenever a document is found in non-Misc workspace, a call is made again to remove it from the Misc workspace.

Reply:
I will not plan on renaming this. Let me know if you have a strong belief that a different name should be chosen here.